bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetAutomationHandler.java

```diff
@@ -0,0 +1,401 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.openwebnet.OpenWebNetBindingConstants;
+import org.openwebnet.message.Automation;
+import org.openwebnet.message.BaseOpenMessage;
+import org.openwebnet.message.Who;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetAutomationHandler} is responsible for handling commands/messages for an Automation OpenWebNet
+ * device. It extends the abstract {@link OpenWebNetThingHandler}.
+ *
+ * @author Massimo Valla - Initial contribution
+ */
+public class OpenWebNetAutomationHandler extends OpenWebNetThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetAutomationHandler.class);
+    private static final SimpleDateFormat formatter = new SimpleDateFormat("ss.SSS");
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.AUTOMATION_SUPPORTED_THING_TYPES;
+
+    protected Automation.Type automationType = Automation.Type.ZIGBEE;
+
+    // internal states
+    public static final int STATE_STOPPED = 0;
+    public static final int STATE_MOVING_UP = 1;
+    public static final int STATE_MOVING_DOWN = 2;
+    public static final int STATE_UNKNOWN = -1;
+
+    // calibration states
+    public static final int CALIBRATION_INACTIVE = -1;
+    public static final int CALIBRATION_ACTIVATED = 0;
+    public static final int CALIBRATION_GOING_UP = 1;
+    public static final int CALIBRATION_GOING_DOWN = 2;
+
+    // positions
+    public static final int POSITION_MAX_STEPS = 100;
+    public static final int POSITION_DOWN = 100;
+    public static final int POSITION_UP = 0;
+    public static final int POSITION_UNKNOWN = -1;
+    public static final int SHUTTER_RUN_UNDEFINED = -1;
+
+    private int shutterRun = SHUTTER_RUN_UNDEFINED;
+    private static final String AUTO_CALIBRATION = "AUTO";
+
+    private long startedMovingAt = SHUTTER_RUN_UNDEFINED;
+    private int internalState = STATE_UNKNOWN;
+    private int positionEst = POSITION_UNKNOWN;
+    private ScheduledFuture<?> moveSchedule;
+    private int positionRequested = POSITION_UNKNOWN;
+    private int calibrating = CALIBRATION_INACTIVE;
+    private static final int STEP_TIME_MIN = 50; // ms
+    private Command commandRequestedWhileMoving = null;
+
+    /// TODO consider making all Automation calls Asynch insted of Synch (blocking), as all the behavior is based on
+    /// received
+    /// state notifications and not on command responses
+
+    public OpenWebNetAutomationHandler(@NonNull Thing thing) {
+        super(thing);
+        logger.debug("==OWN:AutomationHandler== constructor");
```

Comment:
Can these messages be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging